### PR TITLE
CDD-2592_mdg_request

### PR DESCRIPTION
**Pull request by paulblakehmrc**

### CODE REVIEW

File: app/uk/gov/hmrc/customs/declarations/information/xml/BackendPayloadCreator.scala [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+    val searchElement = <n1:labelToRename>{searchType.toString}</n1:labelToRename>.copy(label = searchType.label)
+
+    <n1:queryDeclarationStatusRequest
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xmlns:n1="http://gov.uk/customs/declarationInformationRetrieval/status/v2"
+    xsi:schemaLocation="http://gov.uk/customs/declarationInformationRetrieval/status/v2 queryDeclarationStatusRequest.xsd">
+      <n1:requestCommon>
+        {Seq[NodeSeq](
+        <n1:clientID>99999999-9999-9999-9999-999999999999</n1:clientID>, Text(newLineAndIndentation),
```
Comment (rashadmughal) [inline app/uk/gov/hmrc/customs/declarations/information/xml/BackendPayloadCreator.scala]: Need to pull this out of ASF for CSP and dummy for non-CSP - also does it need to be a VALID UUID?

File: public/api/conf/1.0/application.raml [hunk 2]
```diff
@@ -32,19 +32,168 @@ documentation:
...
+              required: false
+              example: GB123456789000
+          (annotations.scope): "write:customs-declarations-information"
+          securedBy: [ sec.oauth_2_0: { scopes: [ "write:customs-declarations-information" ] } ]
+          responses:
+            200:
+              body:
+                application/xml:
+                  description: Successful response payload
+                  type: !include schemas/wco/declaration/declarationInformationRetrievalStatusResponse.xsd
```
Comment (rashadmughal) [inline public/api/conf/1.0/application.raml]: How do they actually get the schemas? Do we need to provide a zip file like we do in customs-dec?

File: app/uk/gov/hmrc/customs/declarations/information/xml/BackendPayloadCreator.scala (hunk 1 above)
Comment (paulblakehmrc) [reply app/uk/gov/hmrc/customs/declarations/information/xml/BackendPayloadCreator.scala]: Both Errol and John confirmed that neither MDG nor DIS use this value. So for simplicity I left it at a fixed value for both. Yes, it does need to be a uuid. Schema mandates that.

File: public/api/conf/1.0/application.raml (hunk 2 above)
Comment (paulblakehmrc) [reply public/api/conf/1.0/application.raml]: build.sbt create the zip

File: README.md [hunk 3]
```diff
@@ -3,13 +3,16 @@
...
 
-The user must supply a movement reference number (MRN) as a parameter in the request URL. The endpoint returns a set of latest status details for the MRN.
+The user must supply an MRN, DUCR, UCR or inventory reference as a parameter in the request URL. The endpoint returns a set of latest status details.
 
 
 ## GET Status Request 
  ### `GET /mrn/{valid mrn}/status`
+ ### `GET /ducr/{valid ducr}/status`
+ ### `GET /ucr/{valid ucr}/status`
+ ### `GET /inventory-reference/{valid inventoryReference}/status`
```
[the remark this answers is not in the document]
Comment (paulblakehmrc) [reply README.md]: Removed.

File: public/api/conf/1.0/application.raml (hunk 2 above)
Comment (rashadmughal) [reply public/api/conf/1.0/application.raml]: Is there a link to download them?

**Pull request merged**